src/adapter/src/coord/appends.rs

```diff
@@ -352,47 +350,16 @@ impl<S: Append + 'static> Coordinator<S> {
         &mut self,
         timestamp: Timestamp,
         responses: Vec<CompletedClientTransmitter<ExecuteResponse>>,
-        in_ddl: bool,
         _write_lock_guard: Option<OwnedMutexGuard<()>>,
     ) {
         self.apply_local_write(timestamp).await;
-
-        // If we're in the middle of DDL then the catalog and COMPUTE/STORAGE may be out of sync,
-        // so we don't advance other timelines and we don't update read holds.
-        if !in_ddl {
-            let global_timelines = std::mem::take(&mut self.global_timelines);
-            for (
-                timeline,
-                TimelineState {
-                    mut oracle,
-                    mut read_holds,
-                },
-            ) in global_timelines
-            {
-                let now = if timeline == Timeline::EpochMilliseconds {
-                    timestamp
-                } else {
-                    // For non realtime sources, we define now as the largest timestamp, not in
-                    // advance of any object's upper. This is the largest timestamp that is closed
-                    // to writes.
-                    let id_bundle = self.ids_in_timeline(&timeline);
-                    self.largest_not_in_advance_of_upper(&id_bundle)
-                };
-                oracle
-                    .apply_write(now, |ts| self.catalog.persist_timestamp(&timeline, ts))
-                    .await;
-                let read_ts = oracle.read_ts();
-                if read_holds.time.less_than(&read_ts) {
-                    read_holds = self.update_read_hold(read_holds, read_ts).await;
-                }
-                self.global_timelines
-                    .insert(timeline, TimelineState { oracle, read_holds });
-            }
-        }
-
         for response in responses {
             response.send();
         }
+
+        self.internal_cmd_tx
+            .send(Message::AdvanceTimelines)
```

Comment:
We preserve the ordering guarantees (that timeline advancement is processed before any other user-observable thing) because this gets published on `internal_cmd_tx`? Are there even ordering guarantees that we need here?

Reply:
There really are no ordering guarantees that I'm aware of, advancing timelines updates does the following:
1. Update read-holds, which is just to help enable compaction.
2. Advance the read timestamp of non-realtime timelines. Since these are only write-able by upstream sources, we don't have any consistency guarantees with respect to user writes/ddl.

The realtime timeline, where user writes happen, has it's read timestamps explicitly advanced on line 355. We were mainly advancing all timelines here out of convenience. We could even move this message completely out of group commit and create a new interval.

Reply:
I'm fine keeping the message here in that case. Can you pretty much copy that explanation into a comment here? That's the information I wanted to see. LGTM after that.

Reply:
Added.